### ISSUE TITLE
having issue in finetuning .

### ISSUE THREAD

**Issue by pr509** [AI4Bharat/IndicTrans2#54]
i am having issue while finetuning the model. after doing all the preprocessing when i try to run the finetune.sh shell . i am getting an error which shows "**The dataset is empty. This could indicate that all elements in the dataset have been skipped. Try increasing the max number of allowed tokens or using a larger dataset."** .and also can you tell me that should i have to make any changes in the code if i want to do finetuning for only 4 language pair .

**Comment by PranjalChitale** [comment]
Can you confirm if you built sentencepiece from source following the [steps](https://github.com/google/sentencepiece/?tab=readme-ov-file#build-and-install-sentencepiece-command-line-tools-from-c-source) ?

Can you paste the preprocess.log inside the final_bin directory here, that would help us understand if binarization was successful or not.

**Comment by pr509** [comment]
i have created the dataset in same form which the indictrans2 repo suggests.

**Comment by pr509** [comment]
this is the python script i have created .

**Comment by PranjalChitale** [comment]
The script you shared is not accessible.

Also, request you to please read my comment and provide the requisite details I had asked for us to go forward and debug the issue.

**Comment by pr509** [comment]
yes i have build the sentancepiece from source .i am sharing the preprocess.log
[preprocess.log](https://github.com/AI4Bharat/IndicTrans2/files/14738966/preprocess.log)

**Comment by PranjalChitale** [comment]
Can you double check your training data?

From the logs, it appears that you are just trying to train with 4 sentences, and out of that as well half of the tokens are being replaced by unknown tokens.

**Comment by pr509** [comment]
thanks for your help .the issue has been resolved . i just want one help have you trained your model for do not translate can you help you out where could i use do not translate in the model
